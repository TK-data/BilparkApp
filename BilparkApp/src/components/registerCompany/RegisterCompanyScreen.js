import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { connect } from 'react-redux';

import CompanyPicker from './CompanyPicker';
import { getCompanies } from '../../actions/registerCompany';


const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flex: 2,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgb(000, 039, 118)',
  },
  title: {
    color: 'white',
    fontSize: 30,
  },
});

const RegisterCompanyScreen = ({ getCompanyList }) => {
  getCompanyList();

  return (
    <View style={styles.container}>
      <Text style={styles.title}> Velg tilh√∏righet </Text>
      <CompanyPicker />
    </View>
  );
};

RegisterCompanyScreen.navigationOptions = {
  title: 'Selskap',
};

const mapDispatchToProps = (dispatch) => {
  return {
    getCompanyList: () => dispatch(getCompanies()),
  };
};

const mapStateToProps = () => {
  return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(RegisterCompanyScreen);
