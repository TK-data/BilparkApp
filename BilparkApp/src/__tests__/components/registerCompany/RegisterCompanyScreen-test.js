import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import RegisterCompanyScreen from '../../../components/registerCompany/RegisterCompanyScreen';

const initialState = {
};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Testing RegisterCompanyScreen', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <RegisterCompanyScreen />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });
});
