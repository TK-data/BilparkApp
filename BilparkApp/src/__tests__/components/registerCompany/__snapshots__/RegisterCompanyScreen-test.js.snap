// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RegisterCompanyScreen Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RegisterCompanyScreen
    getCompanyList={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 30,
            }
          }
        >
           Velg tilhørighet 
        </Text>,
        <Connect(CompanyPicker) />,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "rgb(000, 039, 118)",
        "display": "flex",
        "flex": 2,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": " Velg tilhørighet ",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "white",
            "fontSize": 30,
          },
        },
        "ref": null,
        "rendered": " Velg tilhørighet ",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
             Velg tilhørighet 
          </Text>,
          <Connect(CompanyPicker) />,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "rgb(000, 039, 118)",
          "display": "flex",
          "flex": 2,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": " Velg tilhørighet ",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "white",
              "fontSize": 30,
            },
          },
          "ref": null,
          "rendered": " Velg tilhørighet ",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
